postgres/datadog_checks/postgres/connections.py

```diff
@@ -51,12 +80,25 @@ def __init__(self, connect_fn):
                 )
         self.connect_fn = connect_fn
 
-    def get_connection(self, dbname, ttl_ms):
+    def _get_connection_raw(self, dbname: str, ttl_ms: int, timeout: int = None) -> psycopg2.extensions.connection:
+        """
+        Return a connection from the pool.
+        """
+        start = datetime.datetime.now()
         self.prune_connections()
         with self._mu:
-            conn = self._conns.pop(dbname, ConnectionWithTTL(None, None))
+            conn = self._conns.pop(dbname, ConnectionInfo(None, None, None, None, None))
             db = conn.connection
             if db is None or db.closed:
+                if self.max_conns is not None and len(self._conns) == self.max_conns:
```

Comment:
nit: the `and len(self._conns) == self.max_conns` condition is different than the while loop, but is actually unnecessary since the while will act like this if statement (bypass the polling if below max connections).

You can also add the `self.max_conns is not None` directly to the while loop, although keeping it as an `if` does help readability of what the loop is doing so I think it's fine to keep the first part.